tests/ios/app/analysis-tests.bzl

```diff
@@ -0,0 +1,100 @@
+load("@bazel_skylib//lib:unittest.bzl", "analysistest", "asserts")
+
+_TestFiles = provider(
+    fields = {
+        "files": "A glob of files collected for later assertions",
+    },
+)
+
+def _identical_outputs_test_impl(ctx):
+    env = analysistest.begin(ctx)
+    all_files = []
+    asserts.true(env, len(ctx.attr.deps) > 1)
+
+    for dep in ctx.attr.deps:
+        if not _TestFiles in dep:
+            continue
+
+        # The input root is the part of the file usually rooted in bazel-out.
+        # For starlark transitions output dirs are fingerprinted by the hash of the
+        # relevant configuration keys.
+        # src/main/java/com/google/devtools/build/lib/analysis/starlark/FunctionTransitionUtil.java
+        for input in dep[_TestFiles].files.to_list():
+            all_files.append(input.root.path)
+
+    # Expect that we have recieved multiple swiftmodules
+    asserts.true(env, len(all_files) > 1)
+
+    # Assert all swiftmodules have identical outputs ( and most importantly an
+    # identical output directory )
+    asserts.equals(env, 1, len(depset(all_files).to_list()))
+    return analysistest.end(env)
+
+def _collect_transitive_outputs_impl(target, ctx):
+    # Collect trans swift_library outputs
+    out = []
+    if ctx.rule.kind == "swift_library":
+        out.extend(target[DefaultInfo].files.to_list())
+
+    if _TestFiles in target:
+        out.extend(target[_TestFiles].files.to_list())
+    if hasattr(ctx.rule.attr, "srcs"):
+        for d in ctx.rule.attr.srcs:
+            if _TestFiles in d:
+                out.extend(d[_TestFiles].files.to_list())
+    if hasattr(ctx.rule.attr, "deps"):
+        for d in ctx.rule.attr.deps:
+            if _TestFiles in d:
+                out.extend(d[_TestFiles].files.to_list())
+    return _TestFiles(files = depset(out))
+
+_collect_transitive_outputs = aspect(
+    implementation = _collect_transitive_outputs_impl,
+    attr_aspects = ["deps", "srcs"],
+)
+
+_identical_outputs_test = analysistest.make(
+    _identical_outputs_test_impl,
+    expect_failure = False,
+    attrs = {
+        "deps": attr.label_list(
+            aspects = [_collect_transitive_outputs],
+        ),
+    },
+)
+
+def make_tests():
+    # This test asserts that transitive dependencies have identical outputs for
```

Comment:
To @gyfelton's question, also curious to know how we'll add more tests here. Should `make_tests` hold all of them?  If that's the case (nit) should the test description be moved into `_identical_outputs_test `?

Reply:
I added a: https://github.com/bazel-ios/rules_ios/pull/196#discussion_r560465686. If you follow the pattern I setup you can add a test for apps / test bundles like:

```
make_tests():

..... 
    _identical_outputs_test(
        name = "test_TestHostEquivilance",
        target_under_test = ":TestAppWithInfoPlist",

        # These inputs *must* be passed seperately, in order to
        # have different transitions applied by Skyframe.
        deps = [":TestAppWithSelectableCopts", ":AppWithSelectableCopts"],
    )
```

but `_identical_outputs_test` cannot be extended as it uses Bazel's analyiss test `analysistest.make`